repo/fsrepo/fsrepo.go

```diff
@@ -400,6 +400,9 @@ func (r *FSRepo) openDatastore() error {
 	} else if r.config.Datastore.Spec == nil {
 		return fmt.Errorf("required Datastore.Spec entry missing from config file")
 	}
+	if r.config.Datastore.NoSync {
+		log.Warning("NoSync is no longer effective. Set 'sync' to false. See https://github.com/ipfs/go-ipfs/blob/master/docs/datastores.md#flatfs.")
```

Comment:
Hrm... I think the wording might be clearer as:
"NoSync is now deprecated in favor of datastore specific settings. If you want to disable fsync on flatfs 'sync' to false,  See <that link>"